python/lsst/ts/ess/common/data_client/tcpip_data_client.py

```diff
@@ -0,0 +1,213 @@
+    simulation_mode : `int`, optional
+        Simulation mode; 0 for normal operation.
+    """
+
+    def __init__(
+        self,
+        config: types.SimpleNamespace,
+        topics: salobj.Controller | types.SimpleNamespace,
+        log: logging.Logger,
+        simulation_mode: int = 0,
+    ) -> None:
+        self.device_configuration: DeviceConfig | None = None
+
+        super().__init__(
+            config=config, topics=topics, log=log, simulation_mode=simulation_mode
+        )
+        self.configure()
+
+        # Lock for TCP/IP communication
+        self.stream_lock = asyncio.Lock()
+
+        self.tcpip_device: TcpipDevice | None = None
+
+    @classmethod
+    def get_config_schema(cls) -> dict[str, typing.Any]:
+        return yaml.safe_load(
+            """
+$schema: http://json-schema.org/draft-07/schema#
+description: Schema for TCP/IP sensors.
+type: object
+properties:
+  host:
+    description: Hostname of the TCP/IP interface.
+    type: string
+    format: hostname
+  port:
+    description: Port number of the TCP/IP interface.
+    type: integer
+    default: 5000
+  connect_timeout:
+    description: Timeout for connecting to the spectrum analyzer (sec).
+    type: number
+  read_timeout:
+    description: >-
+      Timeout for reading data from the sensor (sec).
+    type: number
+  max_read_timeouts:
+    description: Maximum number of read timeouts before an exception is raised.
+    type: integer
+    default: 5
+  name:
+    description: Sensor name.
+    type: string
+  sensor_type:
+    description: Type of the sensor.
+    type: string
+    enum:
+    - CSAT3B
+    - EFM100C
+    - HX85A
+    - HX85BA
+    - LD250
+    - Temperature
+    - Windsonic
+  channels:
+    description: Number of channels.
+    type: integer
+  baud_rate:
+    description: Baud rate of the sensor.
+    type: integer
+    default: 19200
+  poll_interval:
+    description: The poll interval between requests for the telemetry (sec).
+    type: number
+    default: 1.0
+  location:
+    description: Sensor location (used for all telemetry topics).
+    type: string
+required:
+  - host
+  - port
+  - connect_timeout
+  - read_timeout
+  - max_read_timeouts
+  - name
+  - sensor_type
+  - channels
+  - baud_rate
+  - location
+additionalProperties: false
+"""
+        )
+
+    def configure(self) -> None:
+        """Store the device configuration.
+
+        This provides easy access when processing telemetry.
+        """
+        self.device_configuration = DeviceConfig(
+            host=self.config.host,
+            port=self.config.port,
+            name=self.config.name,
+            dev_type=None,
+            dev_id=None,
+            sens_type=self.config.sensor_type,
+            num_channels=self.config.channels,
+            baud_rate=self.config.baud_rate,
+            location=self.config.location,
+        )
+
+    def descr(self) -> str:
+        assert self.tcpip_device is not None  # keep mypy happy
```

Comment:
Do you really want to postface all asserts with "keep mypy happy"? I mean, there's also some value to doing the asserts here besides keeping mypy happy.